acorn/index.d.ts

```diff
@@ -10,13 +10,92 @@ export = acorn;
 import * as ESTree from 'estree';
 
 declare namespace acorn {
-    var version: string;
-    function parse(input: string, options?: Options): ESTree.Program;
-    function parseExpressionAt(input: string, pos: number, options?: Options): ESTree.Expression;
+    interface PlainObject {
+        [name: string]: any;
+    }
+
+    interface Options {
+        ecmaVersion?: 3 | 5 |6 | 7 | 8 | 2015 | 2016 | 2017;
+        sourceType?: 'script' | 'module';
+        onInsertedSemicolon?: (lastTokEnd: number, lastTokEndLoc?: ESTree.Position) => void;
+        onTrailingComma?: (lastTokEnd: number, lastTokEndLoc?: ESTree.Position) => void;
+        allowReserved?: boolean;
+        allowReturnOutsideFunction?: boolean;
+        allowImportExportEverywhere?: boolean;
+        allowHashBang?: boolean;
+        locations?: boolean;
+        onToken?: ((token: Token) => any) | Token[];
+        onComment?: ((
+            isBlock: boolean, text: string, start: number, end: number, startLoc?: ESTree.Position,
+            endLoc?: ESTree.Position
+        ) => void) | Comment[];
+        ranges?: boolean;
+        program?: ESTree.Program;
+        sourceFile?: string;
+        directSourceFile?: string;
+        preserveParens?: boolean;
+        plugins?: {
+            [name: string]: (p: Parser, config: any) => void;
+        };
+    }
+
+    class Parser {
+        constructor(options: Options, input: string, startPos?: number);
+
+        extend(name: string | Symbol, f: (f: Function) => any): void;
+
+        loadPlugins(pluginConfigs: any): void;
+
+        parse(): any;
+
+    }
+
+    const plugin: PlainObject;
```

Comment:
If so, it should be of the type that is currently used for `Options::plugins` and not `any`-based `PlainObject` (`plugins` contains map to actual functions).